commit 9df90cd5da71cbaa18761d12b3c90b65c2906016 (refs/changes/51/3851/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-04-20T21:44:16+00:00 (1 year, 6 months ago)
    
    joyent/node-smartdc-auth#16 KeyPair isKey assertion should be min-versioned

diff --git a/lib/keypair.js b/lib/keypair.js
index bf644ed..4ff597e 100644
--- a/lib/keypair.js
+++ b/lib/keypair.js
@@ -47,7 +47,11 @@ function SDCKeyPair(kr, opts) {
 	this.comment = '';
 
 	if (opts.public !== undefined) {
-		mod_assert.ok(mod_sshpk.Key.isKey(opts.public),
+		/*
+		 * We need a Key of v1,3 or later for defaultHashAlgorithm and
+		 * the Signature hashAlgorithm support.
+		 */
+		mod_assert.ok(mod_sshpk.Key.isKey(opts.public, [1, 3]),
 		    'options.public must be a sshpk.Key instance');
 		this.comment = opts.public.comment;
 	}
